test/sql/aws_secret_external_id.test

```diff
@@ -0,0 +1,30 @@
+# name: test/sql/aws_secret_external_id.test
+# description: test aws secrets with external_id parameter
+# group: [aws]
+
+require no_extension_autoloading "EXPECTED: Test relies on explicit INSTALL and LOAD"
+
+require aws
+
+require httpfs
+
+# Create a secret with external_id parameter
+statement ok
+CREATE SECRET s3_cred (
+    TYPE s3, 
+    PROVIDER credential_chain,
+    chain 'sts',
+    assume_role_arn 'arn:aws:iam::123456789012:role/test-role',
+    external_id 'test-external-id',
+    region 'us-west-2'
+)
+
+# Verify the secret was created
+query I
+SELECT secret_string FROM duckdb_secrets(redact=false) where name='s3_cred';
```

Comment:
Thank you for the test 🙏 . Can you also write one to check the external_id was added? 
Something like 
```
SELECT count(*) FROM duckdb_secrets() where secret_string like '%test-external-id%';
```
should be enough

Reply:
fixed